libs/crosstarget-utils/src/wasm/spawn.rs

```diff
@@ -1,10 +1,23 @@
 use std::future::Future;
 
+use futures::FutureExt;
+use tokio::sync::oneshot;
+use wasm_bindgen_futures::spawn_local;
+
 use crate::common::SpawnError;
 
-pub async fn spawn_if_possible<F>(future: F) -> Result<F::Output, SpawnError>
+pub fn spawn_if_possible<F>(future: F) -> impl Future<Output = Result<F::Output, SpawnError>>
 where
     F: Future + 'static,
 {
-    Ok(future.await)
+    let (sx, rx) = oneshot::channel::<F::Output>();
+    spawn_local(async move {
+        let result = future.await;
+        let _ = sx.send(result);
+    });
+
+    rx.map(|result| match result {
+        Ok(result) => Ok(result),
+        Err(_) => Err(SpawnError),
+    })
```

Comment:
Nit:

```suggestion
    rx.map(|result| result.map_err(SpawnError))
```

Reply:
After bringing `TryFutureExt` to scope as I wrote above, yes.

Reply:
Or, better yet, bring `TryFutureExt` to scope and use `map_err` on the future itself.